modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/application/ApplicationDescriptorFactory.java

```diff
@@ -90,9 +88,10 @@ public ApplicationDescriptor create(File artifactFolder) throws ArtifactDescript
 
       final Set<ArtifactPluginDescriptor> plugins = parsePluginDescriptors(artifactFolder, desc);
       verifyPluginExportedPackages(getAllApplicationPlugins(plugins));
+      desc.setArtifactLocation(artifactFolder);
```

Comment:
There is a rootFolder on the artifact descriptor, that folder points to the folder that contains the exploded artifact file. Can that folder be used instead of the location?